code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Offer.kt

```diff
@@ -0,0 +1,279 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.optimize
+
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.util.DataReader
+import com.adobe.marketing.mobile.util.DataReaderException
+import org.json.JSONObject
+import java.lang.ref.SoftReference
+import java.util.Objects
+
+class Offer private constructor() {
+
+    companion object {
+        private const val SELF_TAG = "Offer"
+
+        @JvmStatic
+        fun fromEventData(data: Map<String, Any?>?): Offer? {
```

Comment:
Please make this internal. It was not a public method in the original implementation